src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -203,41 +204,41 @@ public void updateReplicateStatus(String chainTaskId,
     // TODO: this method needs to be refactored !
     // in case the task has been modified between reading and writing it, it is retried up to 100 times
     @Retryable(value = {OptimisticLockingFailureException.class}, maxAttempts = 100)
-        public void updateReplicateStatus(String chainTaskId,String walletAddress, ReplicateStatus newStatus,
-                                      ReplicateStatusModifier modifier, ReplicateDetails details) {
+        public TaskNotificationType updateReplicateStatus(String chainTaskId, String walletAddress, ReplicateStatus newStatus,
+                                                          ReplicateStatusModifier modifier, ReplicateDetails details) {
         ReplicateStatus currentStatus = null;
         Replicate replicate;
         Optional<ReplicatesList> replicates = getReplicatesList(chainTaskId);
         if (!replicates.isPresent()) {
             log.error(getReplicateStatusErrorLog("replicateList missing",chainTaskId, walletAddress, currentStatus, newStatus, modifier));
-            return;
+            return null;
```

Comment:
Let's use Optional instead of returning null.

Reply:
Thanks!